### PR TITLE
Do not suggest a semicolon for a macro without `!`

### LINKED ISSUE #101490: Invalid help suggestion when `vec` macro incorrectly used without `!`
Given the following code: 

```rust
fn main() {
    let _x = vec[1, 2, 3];
}
```

The current output is:

```
   Compiling playground v0.0.1 (/playground)
error: expected `;`, found `[`
 --> src/main.rs:2:17
  |
2 |     let _x = vec[1, 2, 3];
  |                 ^
  |
help: consider adding `;` here
  |
2 |     let _x = vec;[1, 2, 3];
  |                 +

error: could not compile `playground` due to previous error
```

Just for some context this is the current error message for the stable and beta versions:

```
   Compiling playground v0.0.1 (/playground)
error: expected one of `.`, `?`, `]`, or an operator, found `,`
 --> src/main.rs:2:19
  |
2 |     let _x = vec[1, 2, 3];
  |                   ^ expected one of `.`, `?`, `]`, or an operator

error: could not compile `playground` due to previous error
```
Which is also not ideal as for the other std macros AFAIK (only checked a few), incorrect usage (i.e. without the `!`) will lead to a help suggestion informing the user to include a `!` to invoke a macro (e.g. `print`, `println`).

Hence, the ideal output is to suggest adding the `!` when invoking the `vec` macro incorrectly as with other macros.

@rustbot label +regression-from-stable-to-nightly +D-invalid-suggestion

May need a bisect.

### PR DESCRIPTION
**Pull request by TaKO8Ki**
Fixes a regression in #101490

### CODE REVIEW

File: compiler/rustc_parse/src/parser/expr.rs [hunk 1]
```diff
@@ -1977,6 +1977,9 @@ impl<'a> Parser<'a> {
         open_delim_span: Span,
     ) -> PResult<'a, ()> {
         if self.token.kind == token::Comma {
+            if !self.sess.source_map().is_multiline(prev_span.until(self.token.span)) {
```
Comment (TaKO8Ki) [inline compiler/rustc_parse/src/parser/expr.rs:1980]: This refers to `Parser::expected_one_of_not_found`.

https://github.com/rust-lang/rust/blob/380addd7d2971447d7f6828c508a93fa8018a9b6/compiler/rustc_parse/src/parser/diagnostics.rs#L905-L906

**Comment by Rageking8** [comment]
Sorry just to check if this PR is going to be used to implement the second part of the issue (i.e. suggest adding the `!` when invoking the `vec` macro incorrectly as with other macros). Since this PR seems to be used solely for resolving the regression. Thanks.

**Review by wesleywiser — APPROVED** [review]

**Comment by wesleywiser** [comment]
Thanks @TaKO8Ki!

@bors r+ rollup

**Comment by Rageking8** [comment]
@wesleywiser AFAIK this PR is not supposed to close the referenced issue as the second part is not done. However if wanted I can edit the reference issue to match this PR and make the second part into its own issue. (Note: see my above comment for more context). Thanks.

**Pull request merged**